modules/tinymce/src/core/main/ts/newline/InsertNewLine.ts

```diff
@@ -19,13 +19,21 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>) => {
     () => {
       InsertBr.insert(editor, evt);
       if (Type.isNonNullable(evt)) {
-        fireFakeInputEvent(editor, 'insertLineBreak');
+        const event = fireFakeBeforeInputEvent(editor, 'insertLineBreak');
```

Comment:
Maybe there is some confusion here the branching should be done if there is a event or not. If there is a event passed in then it should fire the before input event to allow the user to prevent the default action, then execute the default action and after that fire a input event. If there is no event passed in it should not fire the beforeInput event or the input event since then it's called from the execCommand. I guess a proper cleanup of this extra parameter should be done after this PR at some point to have two separate functions one that deals with the event and one that deals with direct api calls or something.

- [ ] Branch on the existence of the event and add tests that verify that the default insert action can be prevented if beforeinput is calling preventDefault.

Reply:
Thanks for fixing this up 👍  I also agree we should split this logic at some point, but this is fine for now.